apollo-compiler/src/main/kotlin/com/apollographql/apollo/compiler/HashingAlgorithms.kt

```diff
@@ -0,0 +1,40 @@
+package com.apollographql.apollo.compiler
+
+import java.math.BigInteger
+import java.nio.charset.Charset
+import java.nio.charset.StandardCharsets
+import java.security.MessageDigest
+import java.util.Locale
+
+class HashingAlgorithms(
+    private val algorithm: String?
+) {
+
+    fun applyHashing(text: String): String {
+        return when ((algorithm ?: SHA256).toUpperCase(Locale.ENGLISH)) {
```

Comment:
should this be an enum instead of specific string?

Reply:
Enum would allow for compile time checking so that some of the tests won't be necessary

Reply:
This string comes from project that consumes Apollo library. Not sure how that project can pass enum as gradle parameter.

Reply:
The project can reference classes from the plugin via an import AFAIK.

Reply:
Do you have a snippet that I can refer?